### PR TITLE
Intel: atom: split Baytrail/CherryTrail and Tangier 

### PR DESCRIPTION
**Pull request by plbossart**
We keep having randconfig issues due to the bad partitioning of the Atom code. We really need to have one module for Atom common stuff, and two separate usages of that module for the PCI and ACPI cases.

### CODE REVIEW

File: sound/soc/sof/intel/byt.c [hunk 1]
```diff
@@ -86,606 +44,33 @@ static const struct snd_sof_debugfs_map byt_debugfs[] = {
 	 SOF_DEBUGFS_ACCESS_ALWAYS},
 };
 
-static const struct snd_sof_debugfs_map tng_debugfs[] = {
+static const struct snd_sof_debugfs_map cht_debugfs[] = {
 	{"dmac0", DSP_BAR, DMAC0_OFFSET, DMAC_SIZE,
 	 SOF_DEBUGFS_ACCESS_ALWAYS},
-	{"dmac1", DSP_BAR, DMAC1_OFFSET, DMAC_SIZE,
+	{"dmac1", DSP_BAR,  DMAC1_OFFSET, DMAC_SIZE,
```
Comment (bardliao) [inline sound/soc/sof/intel/byt.c]: Is this intentional?

File: sound/soc/sof/intel/byt.c [hunk 2]
```diff
@@ -86,606 +44,33 @@ static const struct snd_sof_debugfs_map byt_debugfs[] = {
...
 	 SOF_DEBUGFS_ACCESS_ALWAYS},
-	{"ssp0", DSP_BAR, SSP0_OFFSET, SSP_SIZE,
+	{"dmac2", DSP_BAR,  DMAC2_OFFSET, DMAC_SIZE,
```
Comment (bardliao) [inline sound/soc/sof/intel/byt.c]: Here, too. There are double space

**Review by lyakh — APPROVED** [review]

File: sound/soc/sof/intel/byt.c (hunk 2 above)
Comment (plbossart) [reply sound/soc/sof/intel/byt.c]: thanks @bardliao, this wasn't intentional. will fix.

Comment (plbossart) [reply sound/soc/sof/intel/byt.c]: fixed

**Comment by plbossart** [comment]
updated branch (only whitespaces and spurious newline removed)

**Comment by plbossart** [comment]
Need to merge sof-test https://github.com/thesofproject/sof-test/pull/670 first to avoid breaking other PRs.

**Review by bardliao — APPROVED** [review]

**Review by lyakh — APPROVED** [review]

**Comment by plbossart** [comment]
re-doing SOF tests now that PR thesofproject/sof-test#670

**Comment by plbossart** [comment]
SOFCI TEST

**Review by kv2019i — APPROVED** [review]
Thanks, tests now pass as well.

**Pull request merged**